components/supervisor/pkg/terminal/terminal.go

```diff
@@ -94,24 +102,33 @@ func (m *Mux) Close(alias string) error {
 	return nil
 }
 
-func newTerm(pty *os.File, cmd *exec.Cmd) (*term, error) {
+func newTerm(pty *os.File, cmd *exec.Cmd) (*Term, error) {
 	token, err := uuid.NewRandom()
 	if err != nil {
 		return nil, err
 	}
 
-	res := &term{
+	recorder, err := NewRingBuffer(256 << 10)
```

Comment:
very nit: I'd prefer a `const` somewhere at the beginning of the file that sets this value, e.g.
```go
// terminalBacklogSize is the number of bytes of output we'll store in RAM for each terminal.
// The higher this number is, the better the UX, but the higher the resource requirements are.
// For now we assume an average of five terminals per workspace, which makes this consume 1MiB of RAM.
const terminalBacklogSize = 256 << 10
```

Reply:
It also how much data you have to pump through the web socket whenever terminal reconnects.